PWGCF/FemtoDream/TableProducer/femtoDreamProducerTask.cxx

```diff
@@ -733,6 +762,15 @@ struct femtoDreamProducerTask {
       return;
     }
 
+    // Pileup rejection in PbPb data
+    if constexpr (analysePbPb) {
+      if (OptionEvtSpecialSelections.ConfIsUsePileUp &&
+          !colCuts.isPileUpCollisionPbPb(col, OptionEvtSpecialSelections.ConfEvNoSameBunchPileup, OptionEvtSpecialSelections.ConfEvIsGoodITSLayersAll,
+                                         OptionEvtSpecialSelections.ConfTPCOccupancyMin, OptionEvtSpecialSelections.ConfTPCOccupancyMax)) {
+        return;
```

Comment:
In the function there were more things being done but it is only called when checking for pile-up is required. This is confusing and dangerous. Rationalize the function as I commented before and then be consistent in when each condition is checked